nautobot/dcim/tests/test_filters.py

```diff
@@ -4753,4 +4776,71 @@ def test_cable(self):
         self.assertEqual(self.filterset(params, self.queryset).qs.count(), 2)
 
 
+class DeviceRedundancyGroupTestCase(FilterTestCases.FilterTestCase):
+    queryset = DeviceRedundancyGroup.objects.all()
+    filterset = DeviceRedundancyGroupFilterSet
+
+    @classmethod
+    def setUpTestData(cls):
+        common_test_data(cls)
+
+        statuses = Status.objects.get_for_model(DeviceRedundancyGroup)
+        status_active = statuses.get(slug="active")
+        status_planned = statuses.get(slug="planned")
+
+        secrets_groups = list(SecretsGroup.objects.all()[:2])
+
+        DeviceRedundancyGroup.objects.create(
+            name="Device Redundancy Group 1",
+            failover_strategy=DeviceRedundancyGroupFailoverStrategyChoices.FAILOVER_ACTIVE_ACTIVE,
+            status=status_active,
+            secrets_group=secrets_groups[0],
+        )
+
+        DeviceRedundancyGroup.objects.create(
+            name="Device Redundancy Group 2",
+            slug="drg-2",
+            failover_strategy=DeviceRedundancyGroupFailoverStrategyChoices.FAILOVER_ACTIVE_ACTIVE,
+            status=status_planned,
+        )
+
+        DeviceRedundancyGroup.objects.create(
+            name="Device Redundancy Group 3",
+            failover_strategy=DeviceRedundancyGroupFailoverStrategyChoices.FAILOVER_ACTIVE_PASSIVE,
+            status=status_planned,
+            secrets_group=secrets_groups[1],
+        )
```

Comment:
Can we use the groups created by the factory?

Reply:
We could but in fairness it'll probably be the same amount of DB calls as this because I need to create groups of particular conditions for searching: 
- Name
- Slug
- Associated Secrets Group

I figure I'll be searching as much as I am creating but I'll collapse it.